### ISSUE TITLE
New Feature Request . Deploy button

### ISSUE THREAD

**Issue by asadsahi** [mkloubert/vs-deploy#11]
Single click deployment is a good way to trigger, like git sync button. 

Somewhere close to git sync button?

![image](https://cloud.githubusercontent.com/assets/3470892/21347257/b22b9d18-c69f-11e6-982b-5b2f70064835.png)

I love it. :)

**Comment by mkloubert** [comment]
![quick-deploy-demo](https://cloud.githubusercontent.com/assets/4951215/21357062/254fe36a-c6d4-11e6-8ba3-374c8195e7f3.gif)

You have to

* enable the button
* define one or more packages for the button
* define explicit targets in each package

```json
{
    "deploy": {
        "button": {
            "enabled": true,
            "text": "Deploy my stuff!",
            
            "packages": [ "Project 1" ]
        },

        "packages": [
            {
                "name": "Project 1",

                "targets": [
                    "ASP.NET website"
                ]
            }
        ],

        "targets": [
            {
                "name": "ASP.NET website",
                "type": "ftp",
                "host": "ftp.example.com",
                "dir": "/site/wwwroot",
                "user": "myuser", "password": "mypassword",
                "mappings": [
                    {
                        "source": "bin/Debug/netcoreapp1.1/publish",
                        "target": "/"
                    }
                ]
            }
        ]
    }
}
```

**Comment by mkloubert** [comment]
Unfortunately VS Code does not support that feature.

Status bar items are simple buttons, you only can define one single command for each of them.

The only way to get the dictionary of your packages is to press `CTRL + ALT + W` shortcut.

If you [define an explicit target for a package](https://github.com/mkloubert/vs-deploy/issues/8#issuecomment-268211241), as in [the example above](#issuecomment-268279358), you can directly deploy a package and send it to one or more targets (if defined).

**Comment by asadsahi** [comment]
Nice one @mkloubert . Fine by me.

**Comment by asadsahi** [comment]
One minor thing though. When we click on button, output window doesn't open by default.

**Comment by mkloubert** [comment]
Indeed!

**Should be fixed since new available version 3.22.3 now!**
